ast-nodes/routine-call.dart

```diff
@@ -9,13 +12,35 @@ class RoutineCall implements Primary {
 
   RoutineCall(this.name, this.arguments);
 
-  // TODO: implement .parse()
+  factory RoutineCall.parse(Iterable<Token> tokens) {
+    final tempName = tokens.first.value;
+    final iter = tokens.iterator;
+    iter.moveNext(); // Skip the identifier
+    iter.moveNext(); // The opening bracket will always be there, cause it triggers this function to be run
+    var exprs = [];
+    var argExpected =
+        false; // Flag that is set after encountering a comma (to detect cases like func(arg, ))
+    if (!iter.moveNext()) throw SyntaxError(tokens.first, 'Expected ")"');
```

Comment:
Check out [this region](https://github.com/GneyHabub/iCompiler/pull/8/files#diff-ff109be76cd8eb6c38ec6f7ab7979efbR16-R36) (yeah, bragging about my own code). It does largely the same thing, just with different tokens. And I tend to think that one looks more elegant because different concerns in code are separated more (like the one that returns the actual constructed object) + no bool flags cause they ugly. Up for discussion

Reply:
I don't really get how we can get away without using bool flag, cause we don't allow trailing commas. We have to somehow check if we have encountered a comma on the previous run of the cycle. In your example that wasn't the case - every line ends either with \n or ;

Reply:
In my example I `continue` on empty regions, you could throw a SyntaxError instead, so cases like `,,` and trailing commas will be caught